Obsidian.API/Commands/ArgumentParsers/PlayerArgumentParser.cs

```diff
@@ -0,0 +1,25 @@
+﻿namespace Obsidian.API.Commands.ArgumentParsers;
+
+public class PlayerArgumentParser : BaseArgumentParser<IPlayer>
+{
+    public PlayerArgumentParser() : base(7, "minecraft:game_profile")
+    {
+    }
+
+    //TODO support selectors
+    public override bool TryParseArgument(string input, CommandContext context, out IPlayer result)
+    {
+        var server = context.Server;
+        IPlayer? player = null;
+
+        if (Guid.TryParse(input, out var guid))
+            // is valid GUID, try find with guid
+            server.GetPlayer(guid);
```

Comment:
no assignment to `player` variable- this will always return null when a valid GUID is provided.

Reply:
With that, when for whatever crazy reason the player's username is a valid guid...

actually that's not possible I believe but might be good to handle that case anyway

Reply:
Will fix :+1: